_inc/lib/class.core-rest-api-endpoints.php

```diff
@@ -865,6 +871,44 @@ public static function jetpack_connection_status() {
 		);
 	}
 
+	/**
+	 * Test connection status for this Jetpack site. It uses the /jetpack-blogs/%d/test-connection wpcom endpoint.
+	 *
+	 * @since 6.8.0
+	 *
+	 * @return array|WP_Error WP_Error returned if connection test does not succeed. 
+	 */
+	public static function jetpack_connection_test() {
+		$response = Jetpack_Client::wpcom_json_api_request_as_blog(
+			sprintf( '/jetpack-blogs/%d/test-connection', Jetpack_Options::get_option( 'id' ) ),
+			Jetpack_Client::WPCOM_JSON_API_VERSION
+		);
+
+		if ( is_wp_error( $response ) ) {
+			/* translators: %1$s is the error code, %2$s is the error message */
+			return new WP_Error( 'connection_test_failed', sprintf( __( 'Connection test failed (#%1$s: %2$s)', 'jetpack' ), $response->get_error_code(), $response->get_error_message() ), array( 'status' => $response->get_error_code() ) );
+		}
+
+		$body = wp_remote_retrieve_body( $response );
+		if ( ! $body ) {
+			return new WP_Error( 'connection_test_failed', __( 'Connection test failed (empty response body)', 'jetpack' ), array( 'status' => $response->get_error_code() ) );
+		}
+
+		$result = json_decode( $body );
+		$is_connected = (bool) $result->connected;
+		$message = $result->message;
+
+		if ( $is_connected ) {
+			return rest_ensure_response( array(
+				'code' => 'success',
+				'message' => $message,
+				'data' => wp_json_encode( $result ),
```

Comment:
Do we ever use this? Do we need to JSON encode it here or will WP-API do that for us?

Reply:
We don't use it at all! Removed.